wetlab/views.py

```diff
@@ -2441,230 +2410,119 @@ def record_samples(request):
                     },
                 )
-
-            else:
-                # validate types and option lists for projects
-                project_validation = core.utils.samples.validate_project_data(
-                    batch_json_data, project[0], validation
+            # validate sample project data
+            if project_name != "None":
+                (
+                    project_validation,
+                    validation_result,
+                ) = core.utils.samples.validate_project_data(
+                    batch_json_data,
+                    project_name,
                 )
-
-                # If some of the fields are not validated skip to next. DO NOT SAVE
-                for val in project_validation:
-                    if not val["Validate"]:
-                        return render(
-                            request,
-                            "wetlab/record_sample.html",
-                            {
-                                "fields_info": fields_info,
-                                "validation": project_validation,
-                            },
-                        )
-                try:
-                    # Record sample results.
-                    sample_record_result = core.utils.samples.save_recorded_samples(
-                        batch_json_data, req_user, __package__
-                    )
-
-                    # If any samples couldn't be recorded for any error.
-                    for sample in sample_record_result:
-                        if not sample["success"]:
-                            # If some samples got an error. Some samples have been recorded and some of them don't.
-                            return render(
-                                request,
-                                "wetlab/record_sample.html",
-                                {
-                                    "fields_info": fields_info,
-                                    "error_message": "Some samples couldn't be recorded.",
-                                    "sample_record_result": sample_record_result,
-                                },
-                            )
-                except Exception as e:
-                    # In case come uncatched error occurs
-                    error_message = (
-                        "There was an unexpected error when recording the samples"
-                        + str(e)
-                    )
+                if not validation_result:
                     return render(
                         request,
                         "wetlab/record_sample.html",
                         {
-                            "error_message": error_message,
                             "fields_info": fields_info,
+                            "validation": project_validation,
                         },
                     )
-                try:
-                    p_data = core.utils.samples.project_table_fields(project)[0]
```

Comment:
was this failing or something? Why change the function that generates p_data for some code?